### PR TITLE
SDL3 migration

### PR DESCRIPTION
**Pull request by Nanogamer7**
Changes SDL2 to SDL3 (and all associated libraries)

Minor bug introduced with this PR: (link here when merged) sliding along right hand tiles does not work anymore.

Breaking changes:
- Timing is millisecond based, instead of frame/iteration based. Time since last update is passed accordingly.
  - Speed is now increased by a factor of 60
- Most usage of `SDL_Rect` have been replaced with `SDL_FRect`

### CODE REVIEW

File: include/Component.h [hunk 1]
```diff
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <stdint.h>
```
Comment (ezveee) [inline include/Component.h]: shouldn't use c headers
cstdint instead

File: src/SoundManager.cpp
Comment (ezveee) [inline src/SoundManager.cpp:1]: make error messages not rely on Mix_GetError as its not in SDL3

File: src/StatEffectsComponent.cpp
Comment (ezveee) [inline src/StatEffectsComponent.cpp:1]: changes logic of buff duration PlEaSe FiX

File: src/StatEffectsComponent.cpp [hunk 2]
```diff
@@ -32,7 +32,7 @@ void StatEffectsComponent::modifyStatValue(Stats stat, int modifier) //modifier
 	switch (stat)
 	{
 	case Stats::MOVEMENT_SPEED:
-		this->entity->getComponent<TransformComponent>().modifySpeed(modifier);
+		this->entity->getComponent<TransformComponent>().modifySpeed(modifier * 60);
```
Comment (ezveee) [inline src/StatEffectsComponent.cpp]: why exactly 60? pLeAsE eXpLaIn

**Review by ezveee — CHANGES_REQUESTED** [review]

File: src/SoundManager.cpp
Comment (Nanogamer7) [reply src/SoundManager.cpp:1]: apparently `SDL_GetError()` can be used instead

File: src/StatEffectsComponent.cpp (hunk 2 above)
Comment (Nanogamer7) [reply src/StatEffectsComponent.cpp]: changed now required in game implementation

**Review by ezveee — APPROVED** [review]
knuckles approved
all changes understandable and reasonable

**Pull request merged**